### ISSUE TITLE
Beacon does not deactivate when leader terminates the hike

### ISSUE THREAD

**Issue by ItsAdityaKSingh** [CCExtractor/beacon#123]
**Describe the bug**
Beacon does not deactivate when the leader terminates the hike while leaving.

**Expected behavior**
After permission is taken to terminate a hike, it should get terminated.

**Demonstration**

https://user-images.githubusercontent.com/69353350/152857973-1fac4d44-0738-458d-b766-a97d3712f80f.mov

**Are you working on this issue? (Yes/No)**
Yes

**Comment by Aadeesh11** [comment]
Beacon should be active if leader stops location sharing for others to share route. Also if the group is taking a break then in that case leader wants to stop location sharing but does not want to end the hike (which will happen if this is implemented) and will have to create a new beacon altogether. Also there is a direct method to deactivate the beacon.

**Comment by ItsAdityaKSingh** [comment]
I don't think there's a need to pause location sharing or deactivate to again turn on a beacon. The issue is for deactivating after it asks for termination. The issue you are suggesting is for the beacon to be paused or keep running in the background to have a break and stuff. But once the leader exits a hike, and that too after confirming termination, the beacon should end for all.

**Comment by Aadeesh11** [comment]
What Termination currently does is that it stops the location sharing of leader, If you deactivate the beacon then route can't be shared, also if leader wants to pause the beacon, there would'nt be an option. Inshort termination of beacon is , imo, pausing it until resumed or it expires. I think ability to pause is better than deactivating it 😅.

**Comment by ItsAdityaKSingh** [comment]
I guess that's the real meaning of termination and not pausing a hike😅. There's no purpose in pausing a hike if it doesn't stop unless it expires or is deactivated upon the leader's demand. And if a leader is asked and confirms to terminate a hike while leaving it, it must end it for all.

**Comment by Aadeesh11** [comment]
Yes, we can change the change the message to say pause the hike?😅

**Comment by ItsAdityaKSingh** [comment]
Appreciate your views :)

**Comment by Apoorvgarg-creator** [comment]
@ItsAdityaKSingh Can I work on this feature?

**Comment by ItsAdityaKSingh** [comment]
@Apoorvgarg-creator This feature isn't approved for being added. If approved, I would work on it; else, if not able to resolve it wholly, you may take it up :)

**Comment by ItsAdityaKSingh** [comment]
@nb9960 Your views on this feature?
